library/pk_wrap.c

```diff
@@ -653,6 +653,106 @@ static int ecdsa_verify_wrap( void *ctx, mbedtls_md_type_t md_alg,
 }
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+static int pk_ecdsa_sig_asn1_from_psa( unsigned char *sig, size_t *sig_len,
+                                       size_t buf_len );
+
+static int find_ecdsa_private_key( unsigned char **buf, unsigned char *end, int *key_len)
+{
+    size_t len;
+    int ret;
+
+    if( ( ret = mbedtls_asn1_get_tag( buf, end, &len,
+                                      MBEDTLS_ASN1_CONSTRUCTED | MBEDTLS_ASN1_SEQUENCE ) ) != 0 )
+        return( ret );
+
+    /* version */
+    if( ( ret = mbedtls_asn1_get_tag( buf, end, &len,
+                                      MBEDTLS_ASN1_INTEGER ) ) != 0 )
+        return( ret );
+
+    *buf += len;
+
+    /* privateKey */
+    if( ( ret = mbedtls_asn1_get_tag( buf, end, &len,
+                                      MBEDTLS_ASN1_OCTET_STRING ) ) != 0 )
+        return( ret );
+
+    *key_len = len;
```

Comment:
MSVC complains:
```
     2>..\..\library\pk_wrap.c(681): warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data [C:\builds\workspace\mbed-tls-pr-head_PR-5518-head\worktrees\tmpdbrgm5pj\visualc\VS2010\mbedTLS.vcxproj]
```

Normally a size should be `size_t`. We sometimes have to convert sizes to `int` for compatibility with legacy or third-party APIs. In such cases, an explicit cast is often needed, and of course we need to be careful in case the size could overflow (which here, it looks like it can, with invalid input). But here I don't see why `key_len` is an `int *` rather than a `size_t *`.

Reply:
@gilles-peskine-arm thanks for the feedback, I'll fix this and use a `size_t *`